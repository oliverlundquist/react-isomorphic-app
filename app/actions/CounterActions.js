export const INCREMENT_COUNTER = 'INCREMENT_COUNTER'

export const increment = () => ({
	// type: INCREMENT_COUNTER
	type: 'increase'
})

export const incrementIfOdd = () => (dispatch, getState) => {
	const { counter } = getState()

	if (counter % 2 === 0) {
		return
	}

	dispatch(increment())
}

export const incrementAsync = (delay = 1000) => dispatch => {
	setTimeout(() => {
		dispatch(increment())
	}, delay)
}
