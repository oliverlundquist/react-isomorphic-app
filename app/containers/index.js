export { Layout } from './Layout'
export { NotFoundPage } from './NotFoundPage'
export { default as IndexPage } from './IndexPage'
export { default as Users } from './Users'
