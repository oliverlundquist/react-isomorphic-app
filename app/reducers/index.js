export { default as counter } from './CountReducer'
export { default as session } from './SessionReducer'
