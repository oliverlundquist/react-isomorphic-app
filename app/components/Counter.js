import React from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as Components from '../components'
import * as Actions from '../actions'

class Counter extends React.Component {
	render() {
		return (
			<div>
				{this.props.count}<button onClick={this.props.increment}>+</button>
			</div>
		);
	}
}
const mapStateToProps = (state) => ({
	count: state.counter.count
})

function mapDispatchToProps(dispatch) {
	return bindActionCreators({ increment: Actions.increment }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Counter)
