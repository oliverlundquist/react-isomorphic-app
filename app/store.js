import React from 'react';
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { routerReducer } from 'react-router-redux'
import createLogger from 'redux-logger'
import invariant from 'redux-immutable-state-invariant'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'remote-redux-devtools'
import * as reducers from './reducers'
import { reducer as api } from 'redux-json-api-without-credentials'

export const configureStore = (initialState) => {
	// TODO: move logic to external module and refine validation logic
	const environment = process.env.NODE_ENV || 'development'
	const isClient = () => typeof window !== 'undefined' && environment !== 'testing'
	const isServer = () => ! isClient()

	const enhancer = (environment === 'development' && isClient() === true)
		? composeWithDevTools({ realtime: true, port: 3001 })(applyMiddleware(invariant(), createLogger(), thunk))
		: applyMiddleware(thunk)

	return createStore(
		combineReducers({ routing: routerReducer, api: api, ...reducers }),
		initialState,
		enhancer
	)
}
