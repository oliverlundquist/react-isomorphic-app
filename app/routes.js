import React from 'react'
import { Router, Route, IndexRoute } from 'react-router'
import store from './store';
import { Provider } from 'react-redux'
import { Layout, IndexPage, Users, NotFoundPage } from '../app/containers'

const routes = (
	<Route path="/" component={Layout}>
		<IndexRoute component={IndexPage} />
		<Route path="users" component={Users} />
		<Route path="*" component={NotFoundPage} />
	</Route>
)

export default routes;
