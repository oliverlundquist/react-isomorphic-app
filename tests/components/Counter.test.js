import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import { Counter } from '../../app/components'
import { configureStore } from '../../app/store'
import * as Actions from '../../app/actions'

test('Counter should increment when incrementing', () => {
	const store = configureStore({ counter: { count: 5 } })
	const component = renderer.create(
		<Provider store={store}>
			<Counter />
		</Provider>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback
	store.dispatch(Actions.increment());
	store.dispatch(Actions.increment());

	// re-rendering
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
