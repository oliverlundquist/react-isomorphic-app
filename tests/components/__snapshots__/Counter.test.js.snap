exports[`test Counter should increment when incrementing 1`] = `
<div>
  5
  <button
    onClick={[Function]}>
    +
  </button>
</div>
`;

exports[`test Counter should increment when incrementing 2`] = `
<div>
  7
  <button
    onClick={[Function]}>
    +
  </button>
</div>
`;
